krb5/Patch-for-Kerberos-5-1.21.1.patch

```diff
@@ -0,0 +1,2030 @@
++#endif
+ 
+ extern const struct krb5_hash_provider krb5int_hash_md4;
+ extern const struct krb5_hash_provider krb5int_hash_md5;
+diff --git a/src/lib/crypto/krb/etypes.c b/src/lib/crypto/krb/etypes.c
+index fc278783b9..9f7854f50a 100644
+--- a/src/lib/crypto/krb/etypes.c
++++ b/src/lib/crypto/krb/etypes.c
+@@ -103,6 +103,7 @@ const struct krb5_keytypes krb5int_enctypes_list[] = {
+       CKSUMTYPE_HMAC_SHA1_96_AES256,
+       0 /*flags*/, 256 },
+ 
++#ifndef HAVE_LIBWOLFSSL
+     { ENCTYPE_CAMELLIA128_CTS_CMAC,
+       "camellia128-cts-cmac", { "camellia128-cts" },
+       "Camellia-128 CTS mode with CMAC",
+@@ -125,6 +126,7 @@ const struct krb5_keytypes krb5int_enctypes_list[] = {
+       krb5int_dk_cmac_prf,
+       CKSUMTYPE_CMAC_CAMELLIA256,
+       0 /*flags */, 256 },
++#endif
+ 
+     { ENCTYPE_AES128_CTS_HMAC_SHA256_128,
+       "aes128-cts-hmac-sha256-128", { "aes128-sha2" },
+diff --git a/src/lib/krb5/krb/t_etypes.c b/src/lib/krb5/krb/t_etypes.c
+index 90c9f626c6..cc43d0bef5 100644
+--- a/src/lib/krb5/krb/t_etypes.c
++++ b/src/lib/krb5/krb/t_etypes.c
+@@ -61,12 +61,14 @@ static struct {
+         ENCTYPE_DES3_CBC_SHA1, 0 },
+       0, 0
+     },
++#ifndef HAVE_LIBWOLFSSL
+     /* Family with enctype removed */
+     { "camellia -camellia256-cts-cmac",
+       { 0 },
+       { ENCTYPE_CAMELLIA128_CTS_CMAC, 0 },
+       { ENCTYPE_CAMELLIA128_CTS_CMAC, 0 }
+     },
++#endif
+     /* Default set with family added and enctype removed */
+     { "DEFAULT +aes -arcfour-hmac-md5",
+       { ENCTYPE_ARCFOUR_HMAC, ENCTYPE_DES3_CBC_SHA1, 0 },
+diff --git a/src/lib/krb5/os/sendto_kdc.c b/src/lib/krb5/os/sendto_kdc.c
+index 0f4bf23a95..cf0f749f50 100644
+--- a/src/lib/krb5/os/sendto_kdc.c
++++ b/src/lib/krb5/os/sendto_kdc.c
+@@ -1270,9 +1270,9 @@ service_https_write(krb5_context context, const krb5_data *realm,
+         TRACE_SENDTO_KDC_HTTPS_SEND(context, &conn->addr);
+         cm_read(selstate, conn->fd);
+         conn->state = READING;
+-    } else if (st == WANT_READ) {
++    } else if (st == K5_WANT_READ) {
+         cm_read(selstate, conn->fd);
+-    } else if (st == WANT_WRITE) {
++    } else if (st == K5_WANT_WRITE) {
+         cm_write(selstate, conn->fd);
+     } else if (st == ERROR_TLS) {
+         TRACE_SENDTO_KDC_HTTPS_ERROR_SEND(context, &conn->addr);
+@@ -1321,9 +1321,9 @@ https_read_bytes(krb5_context context, struct conn_state *conn,
+     if (st == DONE)
+         return TRUE;
+ 
+-    if (st == WANT_READ) {
++    if (st == K5_WANT_READ) {
+         cm_read(selstate, conn->fd);
+-    } else if (st == WANT_WRITE) {
++    } else if (st == K5_WANT_WRITE) {
+         cm_write(selstate, conn->fd);
+     } else if (st == ERROR_TLS) {
+         TRACE_SENDTO_KDC_HTTPS_ERROR_RECV(context, &conn->addr);
+diff --git a/src/m4/pkg.m4 b/src/m4/pkg.m4
+new file mode 100644
+index 0000000000..1e4694cb34
+--- /dev/null
++++ b/src/m4/pkg.m4
+@@ -0,0 +1,344 @@
++# pkg.m4 - Macros to locate and utilise pkg-config.   -*- Autoconf -*-
```

Comment:
Is this m4 file something that gets auto generated/pulled in? Maybe we can shorten the patch by excluding it.

Reply:
Must have been missing in previous port. Removed and removed some other stuff to cut down the patch.